### PR TITLE
Document string containing JSON as struct

### LINKED ISSUE #1260: Document the support for parsing "string containing JSON" as a struct type
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/11693

### Which part(s) of the docs might be affected or should be updated? And how?

Not sure. Maybe a new topic about parsing data from streams (& create source topics)

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  Update Struct and JSONB pages with tips on how to ingest strings containing JSON formatted data. 

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

    https://github.com/risingwavelabs/risingwave/pull/11693

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1260

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/data-types/data-type-struct.md [hunk 1]
```diff
@@ -44,6 +44,33 @@ CREATE TABLE trip (
...
+    "v1": 10,
+    "v2": "hello"
+  }
+}
+```
+
+You can parse it using the struct type.
+
+```sql
+CREATE SOURCE z (nested STRUCT <v1 INTEGER, v2 VARCHAR> ) FORMAT JSON;
```
Comment (BugenZhao) [inline docs/sql/data-types/data-type-struct.md]:
```suggestion
CREATE SOURCE z (nested STRUCT <v1 INTEGER, v2 VARCHAR>) FORMAT JSON;
```

File: docs/sql/data-types/data-type-struct.md [hunk 2]
```diff
@@ -44,6 +44,33 @@ CREATE TABLE trip (
         );
 ```
 
+You can also use the struct type to parse a string that contains data in JSON format into its proper format.
+
+For instance, if your data is structured like so:
+
+```terminal
```
Comment (BugenZhao) [inline docs/sql/data-types/data-type-struct.md:51]: Should we use `json` here? 👀

File: docs/sql/data-types/data-type-jsonb.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ Notes:
 - Numbers not representable by IEEE 754 double precision floating point may have poor interoperability, notably numbers in the `bigint` type larger than `(2**53)-1`.
 - Avoid using a `JSONB` column for `GROUP BY` and `ORDER BY` clauses or `PRIMARY` and `INDEX` keys. The exact behavior may change in the future.
   - The suggested usage is to extract the target field and cast to a simple type.
+- If you have JSON data contained in a string, use the [Struct](/sql/data-types/data-type-struct.md) type instead of JSONB.
```
Comment (BugenZhao) [inline docs/sql/data-types/data-type-jsonb.md]:
```suggestion
- If you have JSON data contained in a string..
  * use `STRUCT` if the schema is known,
  * or use `VARCHAR` and convert it into `JSONB` later.
```

**Review by BugenZhao — APPROVED** [review]

File: docs/sql/data-types/data-type-struct.md (hunk 2 above)
Comment (hengm3467) [reply docs/sql/data-types/data-type-struct.md:51]: Fixed.

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**